packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -165,26 +268,34 @@ export const SampleFinderSamplesImpl: FC<SampleFinderSamplesGridProps & Injected
 const SampleFinderSamplesWithQueryModels = withQueryModels<SampleFinderSamplesGridProps>(SampleFinderSamplesImpl);
 
 const SampleFinderSamples: FC<SampleFinderSamplesProps> = memo(props => {
-    const { cards, ...gridProps } = props;
-
-    const baseFilters = [];
-    // TODO this is not really correct.  Probably there will be a specialized filter type used for these lineage queries.
-    // cards.forEach(card => {
-    //     if (card.filterArray.length) {
-    //         baseFilters.push(...card.filterArray);
-    //     } else  {
-    //         baseFilters.push(card.entityDataType.inputColumnName.replace("First", card.schemaQuery.queryName), null, Filter.Types.NONBLANK);
-    //     }
-    // });
-
-    const queryConfigs = {
-        allSamples: {
-            id: 'allSamples',
-            title: 'All Samples',
-            schemaQuery: SCHEMAS.EXP_TABLES.MATERIALS,
-            requiredColumns: SAMPLE_STATUS_REQUIRED_COLUMNS,
-        },
-    };
+    const { cards, selectionKeyPrefix, user, ...gridProps } = props;
+    const [queryConfigs, setQueryConfigs] = useState<{ [key: string]: QueryConfig }>(undefined);
+    const [errors, setErrors] = useState<string>(undefined);
+
+    useEffect(() => {
+        (async () => {
+            try {
+                setQueryConfigs(undefined);
+                const configs = await getSampleFinderQueryConfigs(user, cards, selectionKeyPrefix);
+                const promises = [];
+                for (let config of Object.values(configs)) {
+                    promises.push(saveFinderGridView(config.schemaQuery, [{fieldKey: "Name"}]));
```

Comment:
wait, I thought that the hidden grid views were already being saved in the useEffect from `SampleFinderSamplesImpl`?

Reply:
ah ok, got it